components/MyMessage.vue

```diff
@@ -797,6 +810,16 @@ export default {
       const router = useRouter()
       router.push(this.message.type === 'Offer' ? '/give' : '/find')
     },
+    async repostWhenUnavailable() {
+      this.triedToRepost = true
+
+      // see https://github.com/Freegle/iznik-nuxt3/pull/22/files#r1336096366
+      await this.messageStore.fetch(this.id, true)
+      if (this.message.canrepost) {
+        await this.repost()
+        this.triedToRepost = false
+      }
+    },
```

Comment:
@smellyshovel Yes, that's it.  But please:

- Move the reset of `triedToRepost` above the call to `repost` so that it takes effect visually sooner.  
- Remove the comment link to this review comment now
- Add a comment inside the `if` explaining that this can happen if time passes while you stay on the page.

Reply:
@edwh addressed. Marking as resolved.